modules/floating-bars/classes/render/floating-bars-core-render.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+namespace Elementor\Modules\FloatingBars\Classes\Render;
+
+use Elementor\Icons_Manager;
+
+/**
+ * Class Floating_Bars_Core_Render.
+ *
+ * This class handles the rendering of the Floating Bars widget for the core version.
+ *
+ * @since 3.23.0
+ */
+class Floating_Bars_Core_Render extends Floating_Bars_Render_Base {
+
+	protected function render_announcement_icon(): void {
+		$icon = $this->settings['announcement_icon'] ?? '';
+
+		if ( '' !== $icon ): ?>
+			<span class="e-floating-bars__announcement-icon"><?php Icons_Manager::render_icon( $icon, [ 'aria-hidden' => 'true' ] ); ?></span>
+		<?php endif;
+	}
+
+	protected function render_announcement_text(): void {
+		$text = $this->settings['announcement_text'] ?? '';
+		if ( '' !== $text ): ?>
+			<p class="e-floating-bars__announcement-text"><?php echo esc_html( $text ); ?></p>
+		<?php endif;
+	}
+
+	protected function render_cta_icon(): void {
+		$icon = $this->settings['cta_icon'] ?? '';
+
+		if ( '' !== $icon ): ?>
+			<span class="e-floating-bars__cta-icon"><?php Icons_Manager::render_icon( $icon, [ 'aria-hidden' => 'true' ] ); ?></span>
+		<?php endif;
+	}
+
+	protected function render_cta_button(): void {
+		$link = $this->settings['cta_link'] ?? '';
+		$text = $this->settings['cta_text'] ?? '';
+
+		$icon_position = $this->settings['style_cta_icon_position'];
+		$hover_animation = $this->settings['style_cta_button_hover_animation'];
+		$corners = $this->settings['style_cta_button_corners'];
+		$link_type = $this->settings['style_cta_type'];
+		$entrance_animation = $this->settings['style_cta_button_animation'];
+		$entrance_animation_duration = $this->settings['style_cta_button_animation_duration'];
+
+		$cta_classnames = 'e-floating-bars__cta-button';
+
+		if ( ! empty( $hover_animation ) ) {
+			$cta_classnames .= ' elementor-animation-' . $hover_animation;
+		}
+
+		if ( ! empty( $corners ) ) {
+			$cta_classnames .= ' has-corners-' . $corners;
+		}
+
+		if ( ! empty( $link_type ) ) {
+			$cta_classnames .= ' is-type-' . $link_type;
+		}
+
+		if ( ! empty( $entrance_animation ) && 'none' != $entrance_animation ) {
+			$cta_classnames .= ' has-entrance-animation';
+		}
+
+		if ( ! empty( $entrance_animation_duration ) ) {
+			$cta_classnames .= ' has-entrance-animation-duration-' . $entrance_animation_duration;
+		}
+
+		$this->widget->add_render_attribute( 'cta-button', [
+			'class' => $cta_classnames,
+		] );
+
+		if ( ! empty( $link['url'] ) ) {
+			$this->widget->add_link_attributes( 'cta-button', $link );
+			?>
+				<div class="e-floating-bars__cta-button-container">
+					<a <?php echo $this->widget->get_render_attribute_string( 'cta-button' ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?>>
```

Comment:
can we add sanitization here somehow? at least for href attribute to protect from cases like 
`<a href="javascript:alert('Hello, world!');">Click me</a>`

Reply:
Hi @mykytamurzin do you have a suggestion on the best approach to sanitize this? I got this code from the elementor docs: https://developers.elementor.com/docs/editor-controls/control-url/

Reply:
You can try to use `esc_url` here
```
$link = esc_url( $this->settings['cta_link'] ) ?? '';
```
or 
```
$this->widget->add_link_attributes( 'cta-button', esc_url( $link ) );
```

Reply:
@mykytamurzin I don't think this is necessary. Looking at the elementor code, the `add_link_attributes` function already escapes the url:

`$attributes['href'] = esc_url( $url_control['url'], $allowed_protocols );`
(this comes from `includes/base/element-base.php` line 413)

Reply:
good then, thanks for checking